### PR TITLE
[Concept Entry] C++ List

### PR DESCRIPTION
**Pull request by SaviDahegaonkar**
### Description

A new entry on C++ lists

### Issue Solved
#5314 

### Type of Change

- Adding a new entry

### Checklist

- [ ] All writings are my own.
- [ ] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [ ] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/cpp/concepts/list/list.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+The **List** in C++ is a part of the Standard Template Library (STL) and represents a doubly linked list. It allows proficient insertion and deletion of elelments at any position with constant time complexity.
+
+## Syntax
+
+To implement std::list in C++, you need to include the following header:
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:21]: Mention this part as a note

File: content/cpp/concepts/list/list.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+```pseudo 
+#include <list>
+```
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:25]: In this part mention pseudo code for list

File: content/cpp/concepts/list/list.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+myList.remove(10);  // Removes the element with value 10
+// Displaying the updated list
+std::cout << "Updated list elements after deletion: ";
+for (const auto& value : myList) {
+  std::cout << value << " ";  // Output: 5 20
+  }
+std::cout << std::endl;
+  return 0;
+}
+
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:56]: Remove this line

File: content/cpp/concepts/list/list.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+// Removing an element
+myList.remove(10);  // Removes the element with value 10
+// Displaying the updated list
+std::cout << "Updated list elements after deletion: ";
+for (const auto& value : myList) {
+  std::cout << value << " ";  // Output: 5 20
+  }
+std::cout << std::endl;
+  return 0;
+}
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:55]:
```suggestion
#include <iostream>
#include <list>

int main() {
  // Declare a list of integers

  std::list < int > myList;
  // Adding elements to the list

  myList.push_back(10); // Adds 10 to the back
  myList.push_back(20); // Adds 20 to the back
  myList.push_front(5); // Adds 5 to the front
  // Displaying elements in the list
  std::cout << "List elements: ";
  for (const auto & value: myList) {
    std::cout << value << " "; // Output: 5 10 20
  }
  std::cout << std::endl;
  // Removing an element
  myList.remove(10); // Removes the element with value 10
  // Displaying the updated list
  std::cout << "Updated list elements after deletion: ";
  for (const auto & value: myList) {
    std::cout << value << " "; // Output: 5 20
  }
  std::cout << std::endl;
  return 0;
}
```

File: content/cpp/concepts/list/list.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+```
+The output for the above code is:
+
+Here `5`,`10`,and `20` are the numbers that are inserted in the list and also printed the updated list after deletion of `10`.
+
+```shell
+List elements: 5 10 20 
+Updated list elements after deletion: 5 20 
+
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:65]: Remove this blank line

File: content/cpp/concepts/list/list.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+```pseudo 
+#include <list>
+```
+### Example
+
+```cpp
+#include <iostream>
+#include <list>
+int main() {
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:31]: Give a blank line after line 30

File: content/cpp/concepts/list/list.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+std::cout << "Updated list elements after deletion: ";
+for (const auto& value : myList) {
+  std::cout << value << " ";  // Output: 5 20
+  }
+std::cout << std::endl;
+  return 0;
+}
+
+```
+The output for the above code is:
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:58]: Give a blank line after line 50

File: content/cpp/concepts/list/list.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
+---
+Title: 'List' 
+Description: 'In C++, list is a doubly linked list from the STL. It allows efficient insertions and deletions of data'
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:3]:
```suggestion
Description: 'List is a sequential container that stores elements non-contiguous memory locations.' 
```

File: content/cpp/concepts/list/list.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
+---
+Title: 'List'
```
Comment (cigar-galaxy82) [inline content/cpp/concepts/list/list.md:2]:
```suggestion
Title: 'List'
```

**Review by cigar-galaxy82 — CHANGES_REQUESTED** [review]

**Comment by cigar-galaxy82** [comment]
Hi @SaviDahegaonkar thank you for contributing to Codecademy docs, I have suggested a few changes please modify them.

**Comment by mamtawardhani** [comment]
Hi @Sriparno08, quick question: the branch here is incorrect, right? I mean this PR has been raised from the main docs and not the forked repo, correct?
This needs to be closed I guess

**Comment by cigar-galaxy82** [comment]
Hi @SaviDahegaonkar I see what @mamtawardhani mentioned is correct please raise the PR again

**Comment by mamtawardhani** [comment]
@cigar-galaxy82 you can close this PR

**Comment by SaviDahegaonkar** [comment]
Hey @cigar-galaxy82 , Can you close this PR, I am going to re-raise a new PR for this.

**Comment by cigar-galaxy82** [comment]
Closing the PR

**Pull request closed without merging**